test/unit/deadline_client/test_pid_utils.py

```diff
@@ -29,26 +27,78 @@ def test_paths(self):
         location = "/path/to/download/location"
         return {
             "location": location,
-            "pid_file": os.path.join(location, "incremental_output_download.pid"),
+            "pid_file": os.path.join(location, "queue-12345_incremental_output_download.pid"),
         }
 
+    @pytest.fixture
+    def setup_mock_file(self, test_paths, pid="1234"):
+        """
+        Fixture to create and configure a mock file object for testing.
+        """
+        mock_file = MagicMock()
+        mock_file.read.return_value = pid
+        mock_file.fileno.return_value = 5
+        return mock_file
+
+    def setup_context_manager_mock(self, mock_file):
+        """
+        Helper method to create a proper context manager mock for file operations.
+
+        Args:
+            mock_file: The mock file object to be returned by __enter__
+
+        Returns:
+            A mock context manager that returns mock_file when __enter__ is called
+        """
+        mock_context = MagicMock()
+        mock_context.__enter__.return_value = mock_file
+        mock_context.__exit__.return_value = None
+        return mock_context
+
+    def verify_common_assertions(
+        self, mock_open_file, mock_file, pid_file, mock_getsize, mock_realpath
+    ):
+        """
+        Helper method to verify common assertions across tests.
+        """
+        mock_open_file.assert_called_once_with(pid_file, "a+")
+        mock_file.seek.assert_called_once_with(0)
+        mock_file.read.assert_called_once()
+        mock_getsize.assert_called_once_with(pid_file)
+        mock_realpath.assert_called_once_with(pid_file)
```

Comment:
I haven't dug very much into these tests, but I see they're using a lot of mocks. Generally, using mocks this much means that the tests are testing "is the implementation doing what I wrote it to do" which is often not useful and slows down refactoring later. Tests that test "does the implementation satisfy the properties it should" with as few mocks as possible are much better, I recommend you think about how to reduce the amount of mocks your tests have to as close to zero as possible.